File: bit_manipulation/1611_minimum_one_bit_operations_to_make_integers_zero/1611_minimum_one_bit_operations_to_make_integers_zero.py
import math


class Solution:
    def minimumOneBitOperations(self, n: int) -> int:
        if not n:return 0
        head = 1<<int(math.log2(n))
        return head + self.minimumOneBitOperations((n^head)^(head>>1))

n = 5
res = Solution().minimumOneBitOperations(n)
print(res)

File: bit_manipulation/1611_minimum_one_bit_operations_to_make_integers_zero/1611_minimum_one_bit_operations_to_make_integers_zero.go
package main

import "fmt"

func minimumOneBitOperations(n int) int {
	var count, turn int

	for i := 31; i >= 0; i-- {
		if n&(1<<i) > 0 {
			if turn&1 == 0 {
				count += (1 << (i + 1)) - 1
			} else {
				count -= (1 << (i + 1)) - 1
			}
			turn++
		}
	}

	return count
}

func main() {
	n := 5
	res := minimumOneBitOperations(n)
	fmt.Println(res)
}
